File: Lar-Frontend/lib/models/bantuan_program.dart
class BantuanProgram {
  final int id;
  final String title;
  final String description;
  final String? criteria;
  final String status;

  BantuanProgram({
    required this.id,
    required this.title,
    required this.description,
    this.criteria,
    required this.status,
  });

  factory BantuanProgram.fromJson(Map<String, dynamic> json) {
    return BantuanProgram(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      criteria: json['criteria'],
      status: json['status'],
    );
  }
}

File: Lar-Backend/app/Models/BantuanProgram.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class BantuanProgram extends Model
{
    protected $fillable = [
        'title',
        'description',
        'category',
        'criteria',
        'program_type',
        'aid_amount',
        'start_date',
        'end_date',
        'status',
        'admin_id',
        'admin_remarks',
    ];

    protected $casts = [
        'start_date' => 'date',
        'end_date' => 'date',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    // Get the admin who created/updated this program
    public function admin()
    {
        return $this->belongsTo(User::class, 'admin_id');
    }
}
